### DISCUSSION TITLE
Multiple Attachments with the same name - again

### DISCUSSION THREAD

**Discussion by bobskigit** [ravendb/ravendb#13931] [Q&A]
We previously raised a question about multiple attachments with the same name but could not reproduce it. It has happened again.

Screenshot enclosed
![chrome_cJQruDX1KK](https://user-images.githubusercontent.com/4225690/160571571-243cc7df-f684-4a68-bf46-6a17b8932ebb.png)

**Comment by haludi** [comment]
Hi,
Can you export the data and send it to us?

Best Regards,
Igal

**Comment by ayende** [comment]
Also, can you post here the _raw JSON_ for this document? The is the `json` icon that you can click on to get it.

**Comment by bobskigit** [comment]
{

    "Name": "3+",

    "LogoUri": https://www.teleboy.ch/assets/stations/350/icon80_light.png,

    "Owner": "SalesHouses/0000000000000010353-A",

    "BroadcastGroup": "BroadcastGroups/69-B",

    "IsActive": true,

    "ScheduleProvider": {

        "ProviderName": "CHMediaLoader",

        "ExternalRef": "3+"

    },

    "TimeSlots": null,

    "LastUpdatedDateTime": "2022-03-29T08:33:39.3808075+00:00",

    "LastUpdatedUserId": "goldbach",

    ***@***.***": {

        ***@***.***": "Channels",

        "Raven-Clr-Type": "Addecision.Common.Models.Channel, Addecision.Common",

        ***@***.***": "FromReplication, HasAttachments"

    }

}

**Reply by ayende** [reply]
This was editted by _something_ in the middle, it looks like.
Can you post that as a file? There are no `@attachments` here, for example.

**Comment by bobskigit** [comment]
Not sure what you want me to do.

 

I have the specific document open in raven studio.

**Reply by ayende** [reply]
1. Save that file to disk
2. Go to: https://github.com/ravendb/ravendb/discussions/13931#discussioncomment-2459839
3. Update this here as a _file_, not as text content.

**Comment by bobskigit** [comment]
[rawchannel.txt](https://github.com/ravendb/ravendb/files/8370028/rawchannel.txt)

**Reply by ayende** [reply]
Thanks, this is helpful, since I can confirm that this is a bug. I was thinking there may be a whitespace value in the name somehow.
That should have been evident here.

Can you share either the db dump or how to reproduce this?

**Comment by bobskigit** [comment]
I uploaded a dump of the specific collection.

 

I can’t reproduce it in a controlled way, we only notice it when one specific page stops working. This is the code that does the update

 

public async Task UpdateAvatar(string Id, IFormFile avatarImage, string lastUpdatedUserId, string overrideContentType = null)

        {

 

            await using var memoryStream = new MemoryStream();

            await avatarImage.CopyToAsync(memoryStream);

            memoryStream.Seek(0, SeekOrigin.Begin);

 

            // Get content type, not set if uploading from local file

            string contentType = !String.IsNullOrEmpty(overrideContentType) ? overrideContentType : avatarImage.ContentType;

            

            _dbSession.Advanced.Attachments.Store(Id, Avatarkey, memoryStream, contentType);

                   

            // Update last update time

            var channel = await _dbSession.LoadAsync<Channel>(Id);

            channel.LastUpdatedDateTime = DateTimeOffset.Now;

            channel.LastUpdatedUserId = lastUpdatedUserId;

            await _dbSession.StoreAsync(channel);

 

            await _dbSession.SaveChangesAsync();

        }

**Reply by ayende** [reply]
Where did you upload the dump?

**Comment by bobskigit** [comment]
Here is the raw JSON

 

From: ***@***.*** ***@***.***> 
Sent: 29 March 2022 10:26
To: 'ravendb/ravendb' ***@***.***>; 'ravendb/ravendb' ***@***.***>
Cc: 'Author' ***@***.***>
Subject: RE: [ravendb/ravendb] Multiple Attachments with the same name - again (Discussion #13931)

 

Not sure what you want me to do.

 

I have the specific document open in raven studio.

**Comment by bobskigit** [comment]
I replied to the message with an attachment.

**Reply by ayende** [reply]
This goes into GitHub discussions, we cannot see attachments. You need to go to the page directly and add the file, not via email, I'm afraid.
